From 69201821dbae1ca4c3dc20206edb4e3332a0428c Mon Sep 17 00:00:00 2001
From: Jonathan Scott <scottjonathan@google.com>
Date: Tue, 7 May 2019 17:20:00 +0100
Subject: [PATCH 2/2] Make ScreenPinningSettings behaviour consistent with lock
 tasks.

Bug: 127605586
Test: Manual
Change-Id: I4186c2d57b42b6e3d27013f2c9708fca56c561fe
Merged-In: I1cdd82bea4dc9e4182694b061a7746aa83dabdfe
(cherry picked from commit b985aa2f2930e30218bdd1cf1161453b7e622418)
---
 src/com/android/settings/ScreenPinningSettings.java | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/com/android/settings/ScreenPinningSettings.java b/src/com/android/settings/ScreenPinningSettings.java
index fb28dfbfc1..9ff4893303 100644
--- a/src/com/android/settings/ScreenPinningSettings.java
+++ b/src/com/android/settings/ScreenPinningSettings.java
@@ -106,9 +106,11 @@ public class ScreenPinningSettings extends SettingsPreferenceFragment
     }
 
     private boolean isScreenLockUsed() {
-        int def = getCurrentSecurityTitle() != R.string.screen_pinning_unlock_none ? 1 : 0;
+        // This functionality should be kept consistent with
+        // com.android.server.am.ActivityStackSupervisor (see b/127605586)
+        int defaultValueIfSettingNull = mLockPatternUtils.isSecure(UserHandle.myUserId()) ? 1 : 0;
         return Settings.Secure.getInt(getContentResolver(),
-                Settings.Secure.LOCK_TO_APP_EXIT_LOCKED, def) != 0;
+                Settings.Secure.LOCK_TO_APP_EXIT_LOCKED, defaultValueIfSettingNull) != 0;
     }
 
     private boolean setScreenLockUsed(boolean isEnabled) {
-- 
2.17.1

